### ISSUE TITLE
No screen popup when getting a video call

### ISSUE THREAD

**Issue by SoilnRock** [signalapp/Signal-Android#13512]
For a couple of weeks now I'm not getting a popup screen when I get a video call. I have to open signal, go to the contact, select the camara icon and only then/there I can answer the call!
This of course only works when I know who is calling me.

**Comment by alex-signal** [comment]
Could you upload a debug log please?

**Comment by SoilnRock** [comment]
OK, although I haven't done this before, will try to figure out how to do it!

**Comment by SoilnRock** [comment]
Now that was easy!
https://debuglogs.org/android/7.2.4/121f221839fdf8a2c9524256e0c8f47803ca57fce0528911ff4c2162ada58d8a

**Comment by alex-signal** [comment]
When a call comes in, there *should* be a notification that pops up over the top of your screen allowing you to answer the call. Does this not appear either? Is there no notification in your notification shade?

Newer versions of Android utilize CallStyle notifications for incoming calls

**Comment by SoilnRock** [comment]
There's actually _**nothing**_ notifying me in any way. If it wasn't for the ringing I wouldn't even know I was getting a call.
This problem appeared a couple of weeks ago, it was fine before.

**Comment by alex-signal** [comment]
Ok, thanks for confirming! I can see that the IncomingRinger is trying to play back a ringtone. Does your device have incoming calls silenced at all? I'll make sure we're tracking this.

**Comment by SoilnRock** [comment]
No, incoming calls are not silenced and I can hear the ringtone, yet there is no visual notification.
When I have my phone in silent mode I miss very call because there's no visual notification in any way.

**Comment by SoilnRock** [comment]
Man, this is so incredbly frustrating! I'm missing nearly all my calls. People now have to send me a message first that they're plannng on videocalling me. :-(
This really should be fixed ASAP.

**Comment by SoilnRock** [comment]
Is there really no one else having this issue? I'm missing almost all of my calls and it infuriates me so much!

**Comment by alex-signal** [comment]
Hi! We've added some fixes around call notifications in 7.8, which is currently tied up in the Google Review process. One such change includes ensuring vibration works properly when the app is backgrounded. Unfortunately, notification behavior is not always well documented, so it can sometimes be a bit of a guessing game to get things working across all api levels and all devices.

**Comment by SoilnRock** [comment]
Thx for your reply!
The more important thing (for me) would be to actually get a pop up when I receive a call. As desribed earlier, it really takes way to long to unlock phone, go into signal, go into the contact, click on the camera sybbol and only there being able to answer the call. I'm missing 99 % of my calls because of this :-(

**Comment by alex-signal** [comment]
@SoilnRock Still looking into this, fyi. Can you confirm whether this only happens for video calls? I cannot reproduce on my personal Pixel 7 on the latest beta.

Version 7.9 should be going out within the next few days (or early next week) and it would be great if you could try again on that version. I'm sorry this is taking so long! It's sort of like shooting in the dark with these kinds of issues sometimes.

**Comment by SoilnRock** [comment]
A couple of days ago I read a comment on reddit from another guy who had similar problems with his Xiaomi/Redmi:
https://www.reddit.com/r/signal/comments/1d1tnx0/comment/l650n5y/?context=3

I tried the solution from https://www.reddit.com/r/signal/comments/1d1tnx0/comment/l60hu8v/
but at first it didn't work. But nowadays I'm getting a visual popup-screen when someone is videocalling me.
I'm unsure if a restart of the phone was the final step or if Xiaomi had changed something in the meantime.

I could speculate this being a Xiaomi/Redmi specific problem - if that is of any help for further investigation. I had never manually changed anything about the app-permissions, still wonderung why this problem only occured with signalk - anf for example not with Whatsapp calls.

I'm fine now, will come back if that changes - thx for your support!

**Comment by alex-signal** [comment]
It's... very irritating as an Android dev that OEMs feel the need to add stuff like this (there's already a fullscreen permission) 😂 

I'll pass this over to support so maybe we can add something to our help pages.

**Comment by alex-signal** [comment]
@OJ7 Would you be able to share the model name or post a debug log? We are trying to track down phones this occurs on to see if we can come up with something a little more user friendly.
